alacritty_terminal/src/term/mod.rs

```diff
@@ -1840,8 +1846,8 @@ impl<T: EventListener> Handler for Term<T> {
         trace!("Clearing screen: {:?}", mode);
         let template = self.grid.cursor.template;
 
-        // Remove active selections.
-        self.selection = None;
+        let last_line = self.grid.num_lines().0 - 1;
```

Comment:
I'm really conflicted about this variable, can we just go with num_lines here and add the `- 1` wherever it's needed.

That way we don't need to do silly things like `last_line + 1` either, which just evaluates to `num_lines - 1 + 1`.

I've thought about just proposing `last_visible_line`, but even then the indexing direction doesn't make it clear if this is just always `0` or not.

Once that's done this is good to go for me.

Reply:
>I'm really conflicted about this variable, can we just go with num_lines here and add the - 1 wherever it's needed.

It's needed everywhere in this function except one place, that's why it's +1 on when needed, since the places where `+1` is required is basically one, compared to like 5-6 where `-1` is needed. If it was the other way around then sure.

>That way we don't need to do silly things like last_line + 1 either, which just evaluates to num_lines - 1 + 1

But then you have `-1` all over the place.

>I've thought about just proposing last_visible_line, but even then the indexing direction doesn't make it clear if this is just always 0 or not.

It's technically not last visible line, it's last line where cursor can move to, but to name it that way is hard.

Reply:
"Everywhere" is a strong word for three instances of using this versus one of using num_lines. 5-6 is a **strong** exaggeration, that's basically double of the reality. To put it in other words: In 25% of cases you're doing something that makes no sense at all.

> But then you have -1 all over the place.

So? num_lines - 1 is extremely clear on what it means in Alacritty. last_line or last_line + 1 is not at all clear in any way whatsoever.

> It's technically not last visible line, it's last line where cursor can move to, but to name it that way is hard.

It's the last visible line index.